Automattic/jetpack
# Non-admin: Don't require non-admin to be linked to see settings/dashboard

We should not be blocking settings/dashboard access to unlinked non-admins.  This removes the "connected" vs "unconnected" non-admin components, and now only has one "non-admin" view component. 

Since #4766, we have a prompt for unlinked users to connect, so this will suffice as a push to link for now.  

User-level settings such as ATD & PBE do not require the user to be linked, so it's not fair to block access to those if they choose to not link.  

**To Test:** 
- Create a non-admin user on a connected Jetpack site
- Verify you can still see the dashboard/settings page
- Verify that you cannot access admin-only settings 
- Verify that you see the "link to wordpress" notice at the top
- Verify nothing has changed for the admin view. 

cc @eliorivero for review
